src/libambit/device_driver_ambit3.c

```diff
@@ -138,24 +140,79 @@ static int personal_settings_get(ambit_object_t *object, ambit_personal_settings
 
     while (libambit_sbem0102_data_next(&reply_data_object) == 0) {
         switch (libambit_sbem0102_data_id(&reply_data_object)) {
+          case 0x01:
+            settings->date_format = libambit_sbem0102_data_ptr(&reply_data_object)[0];
+            break;
+          case 0x02:
+            settings->tones_mode = libambit_sbem0102_data_ptr(&reply_data_object)[0];
+            break;
+          case 0x03:
+            settings->gps_position_format = libambit_sbem0102_data_ptr(&reply_data_object)[0];
+            break;
+          case 0x04:
+            decli_num = read32(libambit_sbem0102_data_ptr(&reply_data_object), 0);
+            settings->compass_declination_f = *(float*)&decli_num;
```

Comment:
Please make the conversion mathematically explicit.  This type casting voodoo is likely to run foul of strict-aliasing compiler warnings and will break on compilers that use more than four bytes for a float.